presto-spi/src/main/java/com/facebook/presto/spi/connector/ConnectorMetadata.java

```diff
@@ -341,8 +351,15 @@ default void dropView(ConnectorSession session, SchemaTableName viewName)
 
     /**
      * List view names, possibly filtered by schema. An empty list is returned if none match.
+     * Deprecated, please use {@link ConnectorMetadata#listViews(Optional)} instead.
      */
+    @Deprecated
     default List<SchemaTableName> listViews(ConnectorSession session, String schemaNameOrNull)
+    {
+        return listViews(session, Optional.ofNullable(schemaNameOrNull));
```

Comment:
it could be implemented in opposite direction, to make connectors implement the old method and change the engine for now only